teos/cli/teos_cli.py

```diff
@@ -110,85 +155,267 @@ def stop(self):
         print("Closing the Eye of Satoshi")
 
 
-def main(command, args, data_dir, command_line_conf):
-    # Loads config and sets up the data folder and log file
-    config_loader = ConfigLoader(data_dir, CONF_FILE_NAME, DEFAULT_CONF, command_line_conf)
-    config = config_loader.build_config()
+class CliCommand:
+    """
+    Base class of each Cli command.
 
-    setup_data_folder(data_dir)
+    All the implementations should have an appropriately formatted docstring. See existing commands for an example.
+    Any implementation _must_ override the ``name`` attribute, and it might override the ``shortopts`` and ``longopts``
+    attributes.
+    """
 
-    teos_rpc_host = config.get("RPC_BIND")
-    teos_rpc_port = config.get("RPC_PORT")
+    name = None
+    shortopts = ""
+    longopts = []
 
-    rpc_client = RPCClient(teos_rpc_host, teos_rpc_port)
+    @classmethod
+    def parse_args(cls, args):
+        """Parses the ``args`` array using ``getopt``, using ``shortopts`` and ``longopts`` as options."""
 
-    result = None
-    try:
-        if command == "get_all_appointments":
-            result = rpc_client.get_all_appointments()
+        return getopt(args, cls.shortopts, cls.longopts)
```

Comment:
Am I missing something here, or you're not using the opts in any way?

Reply:
Yes, this class is basically meant to be like an abstract class (initially I tried with the `abc` module, but then removed it as all the decorators were just adding tons of additional boilerplate, with little apparent benefit) .
`CLICommand` calls the `parse_args` method and passes the result as an argument to the corresponding `run` method (of the subclass).

Reply:
All the methods of CLICommand and subclasses must be either `static` or `class` method, as no instances are ever created, so there's no value bound to `self`.
`@abstractmethod` would require to use the `abc` package (especially for the CLICommand's `run` method), which I had initially tried, but didn't see to bring any advantage at the end; which is perhaps not surprising as there are no instances aver created, so not much point in calling things "abstract".

I can see why `@classmethod` is typically used for factory methods, as it would give access to the constructors without breaking in case of inheritance (which is impossible if you use a `@staticmethod`); but accessing the class variables from subclasses (`shortopts` and `longopts` here) seems a valid use case as well.